udev/80-azure-disk.rules

```
@@ -34,6 +34,10 @@ ATTRS{nsid}=="?*", ENV{AZURE_DISK_TYPE}="data", PROGRAM="/bin/sh -ec 'echo $$((%
 LABEL="azure_disk_nvme_id"
 ATTRS{nsid}=="?*", IMPORT{program}="/usr/sbin/azure-nvme-id --udev"
 
+LABEL="azure_disk_nvme_set_io_timeout"
+# Azure NVMe disks require 240 second I/O timeout (or greater).
+ENV{DEVTYPE}=="disk", ATTR{queue/io_timeout}="240000"
```

This looks good and is consistent with the settings recommended by the [azure-nvme-preflight-check.sh](https://github.com/Azure/SAP-on-Azure-Scripts-and-Utilities/blob/main/NVMe-Preflight-Check/azure-nvme-preflight-check.sh) script, but I wonder if it's best to set this only on the remote NVME devices (MSFT NVMe Accelerator v1.0) since we have the granular control to do so. Presumably a timeout on a locally attached disk is indicative of an actual probably and we have the opportunity to fail faster with a short timeout there.

---

updated to only affect remote disks